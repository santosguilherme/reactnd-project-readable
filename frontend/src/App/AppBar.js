import React from 'react';
import PropTypes from 'prop-types';

import {withStyles} from 'material-ui/styles';

import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';


const styles = () => ({
    appBar: {
        display: 'flex'
    },
    title: {
        flexGrow: 1
    },
    rightButtonContent: {
        width: '48px',
        marginRight: -12
    },
    rightButton: {},
    leftButtonContent: {
        width: '48px',
        marginLeft: -12
    },
    leftButton: {}
});

const AppBarWrapper = props => {
    const {
        classes,
        leftButton,
        rightButton,
        title
    } = props;


    return (
        <AppBar
            position="static"
            className={classes.appBar}
        >
            <Toolbar>
                <div className={classes.leftButtonContent}>
                    {leftButton && (leftButton)}
                </div>
                <Typography
                    type="title"
                    color="inherit"
                    className={`${classes.title} text-uppercase`}
                    align="center"
                >
                    {title}
                </Typography>
                <div className={classes.rightButtonContent}>
                    {rightButton && (rightButton)}
                </div>
            </Toolbar>
        </AppBar>
    );
};

AppBarWrapper.defaultProps = {
    title: ''
};

AppBarWrapper.propTypes = {
    leftButton: PropTypes.node,
    rightButton: PropTypes.node,
    title: PropTypes.string.isRequired
};

export default withStyles(styles)(AppBarWrapper);