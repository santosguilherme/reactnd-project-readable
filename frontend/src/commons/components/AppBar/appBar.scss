.app-bar {
  display: flex;

  &__title {
    flex-grow: 1;
    text-transform: uppercase;
  }

  &__right-content {
    width: 48px;
    margin-right: -12px;
  }

  &__left-content {
    width: 48px;
    margin-left: -12px;
  }
}