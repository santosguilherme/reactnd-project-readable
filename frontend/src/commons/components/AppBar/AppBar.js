import React from 'react';
import PropTypes from 'prop-types';

import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';

import './appBar.css';


function AppBarWrapper(props) {
    const {leftButton, rightButton, title} = props;

    return (
        <AppBar
            position="static"
            className="app-bar"
        >
            <Toolbar>
                <div className="app-bar__right-content">
                    {leftButton && (leftButton)}
                </div>
                <Typography
                    type="title"
                    color="inherit"
                    className="app-bar__title"
                    align="center"
                >
                    {title}
                </Typography>
                <div className="app-bar__left-content">
                    {rightButton && (rightButton)}
                </div>
            </Toolbar>
        </AppBar>
    );
}

AppBarWrapper.defaultProps = {
    title: ''
};

AppBarWrapper.propTypes = {
    leftButton: PropTypes.node,
    rightButton: PropTypes.node,
    title: PropTypes.string.isRequired
};

export default AppBarWrapper;