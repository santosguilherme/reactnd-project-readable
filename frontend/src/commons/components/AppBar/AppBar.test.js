import React from 'react';
import {shallow} from 'enzyme';

import AppBar from './AppBar';

it('renders without crashing', () => {
    const wrapper = shallow(<AppBar title="test"/>);

    expect(wrapper).toMatchSnapshot();
});

it('Should render left and right buttons', () => {
    const leftButton = <button>Left</button>;
    const rightButton = <button>Right</button>;
    const wrapper = shallow(
        <AppBar
            title="render buttons"
            leftButton={leftButton}
            rightButton={rightButton}
        />
    );

    expect(wrapper).toMatchSnapshot();
});