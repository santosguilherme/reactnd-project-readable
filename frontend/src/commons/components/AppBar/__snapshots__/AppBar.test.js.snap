// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render left and right buttons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBarWrapper
    leftButton={
      <button>
        Left
      </button>
    }
    rightButton={
      <button>
        Right
      </button>
    }
    title="render buttons"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withStyles(Toolbar)>
        <div
          className="app-bar__right-content"
        >
          <button>
            Left
          </button>
        </div>
        <withStyles(Typography)
          align="center"
          className="app-bar__title"
          color="inherit"
          type="title"
        >
          render buttons
        </withStyles(Typography)>
        <div
          className="app-bar__left-content"
        >
          <button>
            Right
          </button>
        </div>
      </withStyles(Toolbar)>,
      "className": "app-bar",
      "position": "static",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="app-bar__right-content"
          >
            <button>
              Left
            </button>
          </div>,
          <withStyles(Typography)
            align="center"
            className="app-bar__title"
            color="inherit"
            type="title"
          >
            render buttons
          </withStyles(Typography)>,
          <div
            className="app-bar__left-content"
          >
            <button>
              Right
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              Left
            </button>,
            "className": "app-bar__right-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Left",
            },
            "ref": null,
            "rendered": "Left",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": "render buttons",
            "className": "app-bar__title",
            "color": "inherit",
            "type": "title",
          },
          "ref": null,
          "rendered": "render buttons",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              Right
            </button>,
            "className": "app-bar__left-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Right",
            },
            "ref": null,
            "rendered": "Right",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withStyles(Toolbar)>
          <div
            className="app-bar__right-content"
          >
            <button>
              Left
            </button>
          </div>
          <withStyles(Typography)
            align="center"
            className="app-bar__title"
            color="inherit"
            type="title"
          >
            render buttons
          </withStyles(Typography)>
          <div
            className="app-bar__left-content"
          >
            <button>
              Right
            </button>
          </div>
        </withStyles(Toolbar)>,
        "className": "app-bar",
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="app-bar__right-content"
            >
              <button>
                Left
              </button>
            </div>,
            <withStyles(Typography)
              align="center"
              className="app-bar__title"
              color="inherit"
              type="title"
            >
              render buttons
            </withStyles(Typography)>,
            <div
              className="app-bar__left-content"
            >
              <button>
                Right
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button>
                Left
              </button>,
              "className": "app-bar__right-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Left",
              },
              "ref": null,
              "rendered": "Left",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": "render buttons",
              "className": "app-bar__title",
              "color": "inherit",
              "type": "title",
            },
            "ref": null,
            "rendered": "render buttons",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button>
                Right
              </button>,
              "className": "app-bar__left-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Right",
              },
              "ref": null,
              "rendered": "Right",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBarWrapper
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withStyles(Toolbar)>
        <div
          className="app-bar__right-content"
        />
        <withStyles(Typography)
          align="center"
          className="app-bar__title"
          color="inherit"
          type="title"
        >
          test
        </withStyles(Typography)>
        <div
          className="app-bar__left-content"
        />
      </withStyles(Toolbar)>,
      "className": "app-bar",
      "position": "static",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="app-bar__right-content"
          />,
          <withStyles(Typography)
            align="center"
            className="app-bar__title"
            color="inherit"
            type="title"
          >
            test
          </withStyles(Typography)>,
          <div
            className="app-bar__left-content"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "app-bar__right-content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": "test",
            "className": "app-bar__title",
            "color": "inherit",
            "type": "title",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "app-bar__left-content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withStyles(Toolbar)>
          <div
            className="app-bar__right-content"
          />
          <withStyles(Typography)
            align="center"
            className="app-bar__title"
            color="inherit"
            type="title"
          >
            test
          </withStyles(Typography)>
          <div
            className="app-bar__left-content"
          />
        </withStyles(Toolbar)>,
        "className": "app-bar",
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="app-bar__right-content"
            />,
            <withStyles(Typography)
              align="center"
              className="app-bar__title"
              color="inherit"
              type="title"
            >
              test
            </withStyles(Typography)>,
            <div
              className="app-bar__left-content"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "app-bar__right-content",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": "test",
              "className": "app-bar__title",
              "color": "inherit",
              "type": "title",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "app-bar__left-content",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
